Wordle.Api/Wordle.Api/Controllers/TokenController.cs

```diff
@@ -0,0 +1,135 @@
+        if (results)
+        {
+            var securityKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_jwtConfiguration.Secret));
+            var credentials = new SigningCredentials(securityKey, SecurityAlgorithms.HmacSha512);
+
+            var claims = new List<Claim>
+            {
+                new Claim(JwtRegisteredClaimNames.Sub, user.UserName!),
+                new Claim(JwtRegisteredClaimNames.Jti, Guid.NewGuid().ToString()),
+                new Claim(Claims.UserId, user.Id.ToString()),
+                new Claim(Claims.Random, (new Random()).NextDouble().ToString()),
+                new Claim(Claims.UserName, user.UserName!.ToString().Substring(0,user.UserName.ToString().IndexOf("@"))),
+            };
+            var roles = await _userManager.GetRolesAsync(user);
+            foreach (var role in roles)
+            {
+                claims.Add(new Claim(ClaimTypes.Role, role));
+            }
+
+            var token = new JwtSecurityToken(
+                issuer: _jwtConfiguration.Issuer,
+                audience: _jwtConfiguration.Audience,
+                claims: claims,
+                expires: DateTime.Now.AddMinutes(_jwtConfiguration.ExpirationInMinutes),
+                signingCredentials: credentials
+            );
+            var jwtToken = new JwtSecurityTokenHandler().WriteToken(token);
+            return Ok(new { token = jwtToken });
+        }
+        return Unauthorized("The username or password is incorrect");
+    }
+
+    [HttpPost("CreateUser")]
+    public async Task<IActionResult> CreateUser([FromBody]CreateUser createUser)
+    {
+        if (string.IsNullOrEmpty(createUser.Username))
+        {
+            return BadRequest("Username is required");
+        }
+        if (string.IsNullOrEmpty(createUser.Password))
+        {
+            return BadRequest("Password is required");
+        }
+        if (string.IsNullOrEmpty(createUser.Name))
+        {
+            return BadRequest("Name is required");
+        }
+        var user = new AppUser
+        {
+            UserName = createUser.Username,
+            Email = createUser.Username,
+            Name = createUser.Name
+        };
+        var result = await _userManager.CreateAsync(user, createUser.Password);
+        if (result.Succeeded)
+        {
+            return Ok();
+        }
+        return BadRequest(result.Errors);
+    }
+
+    [HttpGet("test")]
+    [Authorize]
+    public string Test()
+    {
+        return "something";
+    }
+
+    [HttpGet("testadmin")]
+    [Authorize(Roles=Roles.Admin)]
+    public string TestAdmin()
+    {
+        return "Authorized as Admin";
+    }
+
+    [HttpGet("testruleroftheuniverse")]
+    [Authorize(Roles="RulerOfTheUniverse,Meg")]
+    public string TestRulerOfTheUniverseOrMeg()
+    {
+        return "Authorized as Ruler of the Universe or Meg";
+    }
+
+    [HttpGet("testrandomadmin")]
```

Comment:
We don't need an Admin user for now. It doesn't hurt though, so we can leave it in.